lib/fixture/locator/fixtures.ts

```diff
@@ -52,38 +53,71 @@ const withinFixture: TestFixture<Within, TestArguments> = async (
       : (queriesFor(root, {asyncUtilExpectedState, asyncUtilTimeout}) as WithinReturn<Root>),
   )
 
-declare const queryName: SynchronousQuery
-
-const engine: () => SelectorEngine = () => ({
-  query(root, selector) {
-    const args = JSON.parse(selector, window.__testingLibraryReviver) as unknown as Parameters<
-      Queries[typeof queryName]
-    >
+type SynchronousQueryParameters = Parameters<Queries[SynchronousQuery]>
 
-    if (isAllQuery(queryName))
-      throw new Error(
-        `PlaywrightTestingLibrary: the plural '${queryName}' was used to create this Locator`,
+declare const queryName: SynchronousQuery
+declare class TestingLibraryDeserializedFunction extends DeserializedFunction {}
+
+const engine: () => SelectorEngine = () => {
+  const getError = (error: unknown, matcher: SynchronousQueryParameters[0]) => {
+    if (typeof matcher === 'function' && error instanceof ReferenceError) {
```

Comment:
I think we've accidentally swallowed the error here when the if-block doesn't match

If the `error` is something other than `ReferenceError` then `getError` will return `undefined` and we will throw this at the call site due to `throw getError(error, args[0])`.

Reply:
But I am returning `error` after the `if` ... 😬

Reply:
https://github.com/testing-library/playwright-testing-library/pull/532/files#diff-43d8d50443bceefb7a2a5fafa04c85bc53efb6a9f3786dca414c6a3f21b33d05R77

Reply:
Oh wow, I didn't see that as it was below the "lines removed" part of the diff. 😅

Reply:
Been there! 😆

Reply:
Also, I double-checked, and we have coverage that fails if we swallow errors here.